### PR TITLE
Added bulk actions to Table

### LINKED ISSUE #169: Selecting multiple table rows at once
### Is your feature request related to a problem? Please describe ⚡️
It would be awesome if we could support selecting multiple rows at once. Which makes selecting a range of rows a lot quicker and easier.

### Describe the solution you'd like ✨
We could maybe use the `shift` modifier followed by a click on a checkbox to toggle all checkboxes in between those two.

### PR DESCRIPTION
**Pull request by robinvdv12**
### This PR:

proposed release: `patch`
resolves #169

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)

**Adds** ✨
- `shift-select` option to bulk toggle the Checkboxes.

### CODE REVIEW

File: src/components/Table/index.tsx [hunk 1]
```diff
@@ -27,47 +32,46 @@ const mapAlignment = (alignment: 'left' | 'center' | 'right'): 'flex-end' | 'cen
...
+            toggleAction: true,
+        };
     }
 
     private dragEndHandler = (result: DropResult): void => {
         (this.props.onDragEnd as Function)(result);
     };
 
+    private getItemsInRange = (rows: Array<{ id: string }>, idxOfClickedItem: number): Array<{ id: string }> => {
+        return rows.filter((item, idx): boolean => {
```
Comment (CarloPalinckx) [inline src/components/Table/index.tsx]: idx can be index, we try to avoid abbreviations

File: src/components/Table/index.tsx [hunk 2]
```diff
@@ -27,47 +32,46 @@ const mapAlignment = (alignment: 'left' | 'center' | 'right'): 'flex-end' | 'cen
...
 
-class Table extends Component<PropsType> {
-    private selectedRows: Array<string> = [];
-
+class Table extends Component<PropsType, StateType> {
     public constructor(props: PropsType) {
         super(props);
+
+        this.state = {
+            firstCheckbox: -1,
```
Comment (CarloPalinckx) [inline src/components/Table/index.tsx]: maybe you could rename `firstCheckbox` to something like `selectionStart`,  It seems a bit more explicit to me.

File: src/components/Table/index.tsx [hunk 3]
```diff
@@ -90,23 +94,50 @@ class Table extends Component<PropsType> {
...
+                                        <Row
+                                            key={id}
+                                            alignments={alignments}
+                                            cells={cells}
+                                            draggable={isDraggable}
+                                            selectable={isSelectable}
+                                            index={rowIndex}
+                                            identifier={id}
+                                            onCheck={(event, toggleAction): void => {
+                                                const idxOfClickedItem = this.props.rows.reduce(
```
Comment (CarloPalinckx) [inline src/components/Table/index.tsx]: idx -> index

Comment (CarloPalinckx) [reply src/components/Table/index.tsx]: maybe clicked -> checked as well

File: src/utility/SubscriptionContext/index.tsx [hunk 4]
```diff
@@ -67,6 +69,12 @@ class SubscriptionProvider extends Component<PropsType, StateType> {
         this.setState(prevState => ({ items: prevState.items.filter(item => item.id !== id) }));
     };
 
+    public componentDidUpdate(prevProps: PropsType, prevState: StateType): void {
+        if (this.props.onUpdate !== undefined && JSON.stringify(prevState.items) !== JSON.stringify(this.state.items)) {
+            this.props.onUpdate(this.state.items);
+        }
+
```
Comment (CarloPalinckx) [inline src/utility/SubscriptionContext/index.tsx]: Nit: whitespace

File: src/components/Table/index.tsx [hunk 5]
```diff
@@ -27,47 +32,46 @@ const mapAlignment = (alignment: 'left' | 'center' | 'right'): 'flex-end' | 'cen
...
     }
 
     private dragEndHandler = (result: DropResult): void => {
         (this.props.onDragEnd as Function)(result);
     };
 
+    private getItemsInRange = (rows: Array<{ id: string }>, indexOfCheckedItem: number): Array<{ id: string }> => {
+        return rows.filter((item, idx): boolean => {
+            if (
+                (idx > this.state.selectionStart && idx < indexOfCheckedItem) ||
```
Comment (CarloPalinckx) [inline src/components/Table/index.tsx]: idx -> index

File: src/components/Table/index.tsx [hunk 6]
```diff
@@ -27,47 +32,46 @@ const mapAlignment = (alignment: 'left' | 'center' | 'right'): 'flex-end' | 'cen
...
+            toggleAction: true,
+        };
     }
 
     private dragEndHandler = (result: DropResult): void => {
         (this.props.onDragEnd as Function)(result);
     };
 
+    private getItemsInRange = (rows: Array<{ id: string }>, indexOfCheckedItem: number): Array<{ id: string }> => {
+        return rows.filter((item, idx): boolean => {
```
Comment (CarloPalinckx) [inline src/components/Table/index.tsx]: idx -> item

Comment (robinvdv12) [reply src/components/Table/index.tsx]: do you mean `idx -> index`?
because `idx ->  item` makes no sense to me.

**Pull request merged**